compiler/src/lnntoamm/Types.ts

```diff
@@ -300,28 +721,36 @@ class Generated extends Type {
   instance(): Type {
     if (this.delegate !== null) {
       return this.delegate.instance();
+    } else if (this.tempDelegate !== null) {
+      return this.tempDelegate.instance();
     } else {
       return new Generated();
     }
   }
 
-  tempConstrain(to: Type) {
+  tempConstrain(to: Type, scope: Scope) {
     if (this.delegate !== null) {
-      this.delegate.tempConstrain(to);
+      this.delegate.tempConstrain(to, scope);
     } else if (this.tempDelegate !== null) {
       TODO('temp constraints to a temporary constraint???');
+    } else if (to instanceof Has) {
+      TODO("i'm not sure");
     } else {
       this.tempDelegate = to;
     }
   }
 
   resetTemp() {
-    if (this.tempDelegate !== null) {
+    if (super.tempDelegate !== null) {
```

Comment:
Is this change right? If so, why doesn't the `super` also get its `tempDelegate` removed?

Reply:
HA, SO

I found out that, apparently, JS's `super` only works on functions :) I don't know why, and MDN doesn't explicitly state this, but whenever you do `super.<field>`, it's actually `undefined`. This error just never triggered because it's impossible to have a `Generated`-typed `FnParam`

Reply:
I was actually banging my head against the wall for at least an hour because I was trying to access `super.tempDelegate` in the other methods and it was never working and I had no idea why and then I decided to try accessing with `this` and it worked and I just... *sigh*